src/System.Drawing.Common/src/System/Drawing/GdiplusNative.Unix.cs

```diff
@@ -94,6 +95,33 @@ internal static IntPtr LoadNativeLibrary()
             private static void PlatformInitialize()
             {
                 LibraryResolver.EnsureRegistered();
+
+                // Check whether the version of libgdiplus is a recent version of libgdiplus.
+                var installedVersion = LibgdiplusVersion;
+
+                if (installedVersion == null)
+                {
+                    throw new PlatformNotSupportedException(SR.Format(SR.LibgdiplusOutdated, LibraryRequiredVersion));
+                }
+                else if (installedVersion < LibraryRequiredVersion)
+                {
+                    throw new PlatformNotSupportedException(SR.Format(SR.LibgdiplusOutdated2, LibraryRequiredVersion, installedVersion));
+                }
+            }
+
+            private static Version LibgdiplusVersion
```

Comment:
Is it really worth it for this to be separate function?

```
Version installedVersion;

try
{
    installedVersion = new Version(GetLibgdiplusVersion());
}
catch (EntryPointNotFoundException)
{
    // in case of older libgdiplus
    throw new PlatformNotSupportedException(SR.Format(SR.LibgdiplusOutdated, LibraryRequiredVersion));
}

if (installedVersion < LibraryRequiredVersion)
{
    throw new PlatformNotSupportedException(SR.Format(SR.LibgdiplusOutdated2, LibraryRequiredVersion, installedVersion));
}
```